metecho/api/serializers.py

```diff
@@ -48,6 +49,48 @@ def to_representation(self, value):
         return str(value.pk)
 
 
+class GitHubIssueSerializer(serializers.ModelSerializer):
+    id = serializers.CharField(read_only=True)
+    project = HashidPrimaryKeyRelatedField(queryset=Project.objects.all())
+    epic = serializers.SerializerMethodField()
+    task = serializers.SerializerMethodField()
+
+    class Meta:
+        model = GitHubIssue
+        read_only_fields = (
+            "id",
+            "number",
+            "title",
+            "created_at",
+            "html_url",
+            "project",
+            "epic",
+            "task",
+        )
+        fields = read_only_fields
+
+    def get_epic(self, issue):
+        try:
+            return {
+                "id": str(issue.epic.id),
+                "name": issue.epic.name,
+                "status": issue.epic.status,
+            }
+        except Epic.DoesNotExist:
+            return None
+
+    def get_task(self, issue):
+        try:
+            return {
+                "id": str(issue.task.id),
+                "name": issue.task.name,
+                "status": issue.task.status,
+                "review_status": issue.task.review_status,
+            }
```

Comment:
Looking at the mockup, I see why you nested these here. To generate the link to the Task Detail page, though, I think we'll also need the `issue.task.epic.slug` and `issue.task.slug`. (Unless we just want the `id` and make the front end look up the necessary info.)

Reply:
Would you say nesting is better in this case or we should just return the epic/task ID?

Reply:
For now fixed in 94e344587db6a311467b76c6be3f1831c8a2c96f

Reply:
I like this for now -- unless we refactor, we aren't guaranteed to have the full list of Tasks in the front end store at this point. 👍